presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```diff
@@ -383,24 +383,31 @@ private ImmutableMap<VariableReferenceExpression, ColumnHandle> createAssignment
                             if (!partitionField.transform().isIdentity()) {
                                 Type partitionFieldType = partitionField.transform().getResultType(sourceField.type());
                                 VariableReferenceExpression variableReference = variableAllocator.newVariable(partitionField.name(), toPrestoType(partitionFieldType, typeManager));
-                                IcebergColumnHandle columnHandle = new IcebergColumnHandle(
-                                        ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
-                                        toPrestoType(partitionFieldType, typeManager),
-                                        Optional.empty(),
-                                        PARTITION_KEY);
-                                unselectedAssignmentsBuilder.put(variableReference, columnHandle);
+                                if (!selectedFields.contains(partitionField.fieldId())) {
```

Comment:
How are you able to reach this case where the partition field is in the selected fields? My understanding is that if the partition field is not an identity field you can't query it. 

Can you share the query statement and the table definition?

Reply:
If this is related to the `.add` to the selected fields that happens below than this condition can be removed since adding the sourceField id will never match the partition field id of a non identity field (even for identity partitions the field id is different)